### PR TITLE
DISABLE_TRANSACTION_BROADCAST logic moved to tests suites

### PR DESCRIPTION
**Pull request by bharamboure-ledger**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - ...

### 📝 Description

`DISABLE_TRANSACTION_BROADCAST` logic moved from github workflow to tests suites 

### ❓ Context

- **JIRA or GitHub link**: 

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: apps/ledger-live-desktop/tests/specs/speculos/delegate.spec.ts [hunk 1]
```diff
@@ -47,17 +47,121 @@ const validators = [
   },
 ];
 
-for (const account of e2eDelegationAccounts) {
-  test.describe("Delegate", () => {
+test.describe("Delegate flows", () => {
+  test.beforeAll(async () => {
+    if (new Date().getDay() === 1) {
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-desktop/tests/specs/speculos/delegate.spec.ts:52]: but this means that any execution that happens on day 1 will have ENABLE_TRANSACTION_BROADCAST = true, and we don't want this, we want only broadcast once during nightly and that's it, also for release validation we need to have ability to include/enable broadcasting no matter the day

Comment (abdurrahman-ledger) [inline apps/ledger-live-desktop/tests/specs/speculos/delegate.spec.ts:52]: Shouldn't we also add a condition to broadcast only scheduled run or a condition to do on demand for release ?
(Same for all below)

Comment (bharamboure-ledger) [reply apps/ledger-live-desktop/tests/specs/speculos/delegate.spec.ts]: indeed, added an additional check
will test tomorrow

Comment (bharamboure-ledger) [reply apps/ledger-live-desktop/tests/specs/speculos/delegate.spec.ts]: indeed, added an additional check
will test tomorrow

File: apps/ledger-live-desktop/tests/specs/speculos/delegate.spec.ts [hunk 2]
```diff
@@ -47,17 +48,120 @@ const validators = [
   },
 ];
 
-for (const account of e2eDelegationAccounts) {
-  test.describe("Delegate", () => {
+test.describe("Delegate flows", () => {
+  test.beforeAll(async () => {
+    process.env.ENABLE_TRANSACTION_BROADCAST =
+      new Date().getDay() === 1 && isRunningInScheduledWorkflow() ? "true" : "false";
```
Comment (abdurrahman-ledger) [inline apps/ledger-live-desktop/tests/specs/speculos/delegate.spec.ts:54]: Should be `new Date().getDay() === 1 && isRunningInScheduledWorkflow() ? "1" : "0";` no ?

Comment (bharamboure-ledger) [reply apps/ledger-live-desktop/tests/specs/speculos/delegate.spec.ts:54]: I think both work tbh, will update still

Comment (abdurrahman-ledger) [reply apps/ledger-live-desktop/tests/specs/speculos/delegate.spec.ts]: https://github.com/LedgerHQ/ledger-live/blob/3eee97630e774b2ad38b8427f0328d0c77226fcb/apps/ledger-live-desktop/tests/fixtures/common.ts#L156

We're checking if  equals "1"

**Review by abdurrahman-ledger — APPROVED** [review]

**Pull request merged**